File: client/src/types/analytics.ts
export interface CategorySpend {
  category_name: string;
  icon: string;
  color: string;
  total_amount: number;
  percentage: number;
}

export interface DashboardSummary {
  total_monthly_recurring: number;
  pending_bills_count: int;
  pending_bills_total: number;
  active_subscriptions_count: number;
  next_critical_due_date?: string | null;
  category_breakdown: CategorySpend[];
}

export interface FinancialInsight {
  monthly_outlook: string;
  actionable_tips: string[];
}

File: server/app/schemas/analytics.py
from datetime import date
from decimal import Decimal
from typing import List, Optional
from pydantic import BaseModel


class CategorySpend(BaseModel):
    category_name: str
    icon: str
    color: str
    total_amount: Decimal
    percentage: float


class MonthlySpendTrend(BaseModel):
    month: str  # e.g., "Jan", "Feb"
    year: int
    bills_amount: Decimal
    subscriptions_amount: Decimal
    total: Decimal


class DashboardSummaryResponse(BaseModel):
    total_monthly_recurring: Decimal
    pending_bills_count: int
    pending_bills_total: Decimal
    active_subscriptions_count: int
    next_critical_due_date: Optional[date] = None
    category_breakdown: List[CategorySpend] = []
